src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,32 +26,81 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+        private static readonly TimeSpan SrvLookupTimeout = TimeSpan.FromSeconds(30);
+
+        public ConfigurationClientManager(
+            IEnumerable<string> connectionStrings,
+            ConfigurationClientOptions clientOptions,
+            bool replicaDiscoveryEnabled)
+        {
             if (connectionStrings == null || !connectionStrings.Any())
             {
                 throw new ArgumentNullException(nameof(connectionStrings));
-            }
-
-            _clients = connectionStrings.Select(connectionString => 
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
+            }
+
+            string connectionString = connectionStrings.First();
+            _endpoint = new Uri(ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.EndpointSection));
+            _secret = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.SecretSection);
+            _id = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.IdSection);
+            _clientOptions = clientOptions;
+            _replicaDiscoveryEnabled = replicaDiscoveryEnabled;
+
+            _srvLookupClient = new Lazy<SrvLookupClient>();
+
+            _clients = connectionStrings
+                .Select(cs =>
+                {
+                    var endpoint = new Uri(ConnectionStringUtils.Parse(cs, ConnectionStringUtils.EndpointSection));
+                    return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(cs, _clientOptions));
+                })
+                .ToList();
+        }
+
+        public ConfigurationClientManager(
+            IEnumerable<Uri> endpoints,
+            TokenCredential credential,
+            ConfigurationClientOptions clientOptions,
+            bool replicaDiscoveryEnabled)
+        {
             if (endpoints == null || !endpoints.Any())
             {
                 throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
-        }
+            }
+
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+
+            _endpoint = endpoints.First();
+            _credential = credential;
+            _clientOptions = clientOptions;
+            _replicaDiscoveryEnabled = replicaDiscoveryEnabled;
+
+            _srvLookupClient = new Lazy<SrvLookupClient>();

```

Comment:
what's the value of Lazy here?

Reply:
The value is not yet created here, it's null I believe.

Reply:
It's not okay to just do `_srvLookupClient = SrvLookupClient;`?